blocks.js

```diff
@@ -3000,6 +3068,129 @@ BlockMorph.prototype.codeMappingHeader = function () {
     return hat;
 };
 
+// Variable refactoring
+
+BlockMorph.prototype.refactorThisVar = function (justTheTemplate) {
+    // Rename all occurrences of the variable this block is holding,
+    // taking care of its lexical scope
+
+    var myself = this,
+        oldName = this.blockSpec,
+        receiver = this.receiver(),
+        ide = this.parentThatIsA(IDE_Morph),
+        stage = ide.stage,
+        oldWatcher = receiver.findVariableWatcher(oldName),
+        oldValue, newWatcher;
+
+    new DialogBoxMorph(
+            this,
+            function (newName) {
+                var definer;
+                
+                if (this.parent instanceof SyntaxElementMorph) {
```

Comment:
I think it doesn't matter much, as `DialogBoxMorph`s execute their action in the context of the specified target:

    if (typeof this.target === 'function') {
        if (typeof this.action === 'function') {
            this.target.call(this.environment, this.action.call());
        } else {
            this.target.call(this.environment, this.action);
        }
    } else {
        if (typeof this.action === 'function') {
            this.action.call(this.target, this.getInput());
        } else { // assume it's a String
            this.target[this.action](this.getInput());
        }
    }

So, when you instantiate one with:

    new DialogBoxMorph(this, [...])

The target is going to be the context where it's been instantiated.

@jmoenig did I get that right?